### ISSUE TITLE
`f32::round` fails to link with emscripten (llvm_round_f32 not found)

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust#37185]
``` rust
fn main() {
    let float = 0.5f32;
    float.round();
}
```

Compile with `rustc test.rs --target=asmjs-unknown-emscripten` and emcc will abort with an error:

```
error: unresolved symbol: llvm_round_f32
```

**Comment by jonas-schievink** [comment]
It is also complaining about unexpected EOF:

```
/home/jonas/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /home/jonas/.multirust/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/asmjs-unknown-emscripten/lib/libcompiler_builtins-e428224f6caf212a.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered
```

This happens for every rlib in the standard library, but I'm assuming this is just spurious and will go away when the real bug is fixed.

**Comment by jeffparsons** [comment]
I'm seeing only the latter in my experimentation:

```
[jeff@host:~/Projects/planetkit]$ "emcc" "-L" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/planetkit.0.o" "-o" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/planetkit.js" "-Wl,--gc-sections" "-nodefaultlibs" "-L" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps" "-L" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/build/miniz-sys-60c8d67696f63a43/out" "-L" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib" "-Wl,-Bstatic" "-Wl,-Bdynamic" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libopengl_graphics-01d368bad46d051c.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libpiston_window-f1a7339ccd920121.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libshaders_graphics2d-1bd8b10e21509474.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_device_gl-f3e972e45450a75d.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_gl-b7e8551f5d537140.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libcamera_controllers-994144fde7c965e8.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libcam-11900c50c377715f.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libquaternion-c4b4a533df0e2749.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libpiston-eedbb5dfb932e300.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libfnv-1364d96376ddfb32.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_graphics-e1dadcda68636ea4.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/librusttype-c21162edb59b345b.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/liblinked_hash_map-28a664163baff857.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libarrayvec-682f4bdc47ddb22e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnodrop-8aef2816a14c0a9e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libodds-9e106c1ee0ec95f0.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libshaders_graphics2d-574001b4272bccea.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgraphics-bf08eea27a89d49e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libinterpolation-26420a82c583f0b1.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libread_color-eb34d097612ac965.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libvecmath-91e4c22bc2d512af.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libstb_truetype-159d3ced5bac15cb.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libbyteorder-7a494f72a43262ac.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libglutin_window-7ca4ce7285cb65ad.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgl-f86824b65a1d579a.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libglutin.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libshared_library-94c5183073084fbf.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/liblazy_static-359f5533c970cd71.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libevent_loop-732c3c9fe22b8e20.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libwindow-8a4b85c2becb3422.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libinput-2eeadab8b08583f9.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libviewport-d1b29131e8c075c9.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libfloat-5f568cf52eab6ad2.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libshader_version-b8072d7c2b12472d.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_texture-1b3ab46d4fd1c4b1.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libimage-1ef1c5676f65109e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libscoped_threadpool-c9a9d52c8af25eca.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgif-9f13a06f8f49267e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libcolor_quant-ccb3bfc104102a50.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/liblzw-6ef9d38d31cf3d1c.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libjpeg_decoder-e7f902581e17f586.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/librayon-c3557c1c242af173.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_cpus.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libdeque-82214e5f75d78bdf.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libpng-47615b6a63061b53.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libflate2-d719035eaa7c6a88.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libminiz_sys-722889de4af2439c.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libinflate-3b154d39385bf8a5.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libenum_primitive-77b731c79b5cdf4e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum-1fa4854b44bb6a54.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_iter-50df698bc905252c.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libbyteorder-e4fc6ea0c49cabcf.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libtexture-e75555caa875ee44.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx-a301f0b8752d5663.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_core-c138efaf774bac65.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libbitflags-0e272044714c8076.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libdraw_state-196f0a026baef11a.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libbitflags-b8c7fb7df9b2bc2e.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/liblog-bf16bb9a4912b11d.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_rational-7726582d22be0b52.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_bigint-a144dbec313e5e23.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_integer-52fdddf28cd8e924.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libnum_traits-92bb90166cd1857c.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/librand-49a08859d086fffe.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/liblibc-1417726cb94dbc83.rlib" "/Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/librustc_serialize-3561541d79c18212.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libstd-17342542cc541012.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libpanic_unwind-f78756b576499725.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libunwind-11f7709e0c71505b.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/librand-46ed9b788a6928f6.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libcollections-e32369d7fef31fbf.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/liballoc-24699c1ddb055eb0.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/liballoc_system-3e467e865c8fa572.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/liblibc-60365c932e50e382.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/librustc_unicode-844a33a197b559a5.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libcore-d9873b515905cac5.rlib" "/Users/jeff/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/wasm32-unknown-emscripten/lib/libcompiler_builtins-e428224f6caf212a.rlib" "-l" "c" "-s" "BINARYEN=1" "-s" "ERROR_ON_UNDEFINED_SYMBOLS=1"
/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libopengl_graphics-01d368bad46d051c.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libpiston_window-f1a7339ccd920121.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libshaders_graphics2d-1bd8b10e21509474.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_device_gl-f3e972e45450a75d.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libgfx_gl-b7e8551f5d537140.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/jeff/Downloads/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/jeff/Projects/planetkit/target/wasm32-unknown-emscripten/debug/deps/libcamera_controllers-994144fde7c965e8.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered
```

(Etc.)

I have local path overrides for `num_cpus` and `glutin`.

**Comment by jeffparsons** [comment]
I'm running rustc 1.14.0-nightly (098d22845 2016-10-13).

**Comment by tomaka** [comment]
The missing symbol and the end of file errors should probably be two different issues.

> error: unresolved symbol: llvm_round_f32

The symbol should be added to emscripten, since it doesn't seem to have it.
See [this PR](https://github.com/kripken/emscripten/pull/4095/files) as an example of how to do it.

**Comment by badboy** [comment]
Fixed upstream: https://github.com/kripken/emscripten/pull/4662

**Comment by Ostrovski** [comment]
So, what about `The end of the file was unexpectedly encountered` part? Having plenty of such warnings while trying to complile my app that uses rust-sdl2 with emscripten. 

```
emcc -L /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/gwar-4bbeb8b6e0396b50.0.o -o /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/gwar-4bbeb8b6e0396b50.js -Wl,--gc-sections -nodefaultlibs -L /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps -L /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib -Wl,-Bstatic -Wl,-Bdynamic /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libsdl2-a3439967d6cd896a.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libsdl2_sys-6a6cdeed38de6269.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/liblazy_static-5e13a0eed191ed30.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/librand-35167bebb3b301e7.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/liblibc-09f884e06faabf22.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum-351b1cddc85503c1.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_iter-618824a5957aaa8f.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_integer-9eb11829964f3c6e.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_traits-b3505e86c388a8aa.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libstd-17342542cc541012.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libpanic_unwind-f78756b576499725.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libunwind-11f7709e0c71505b.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/librand-46ed9b788a6928f6.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcollections-e32369d7fef31fbf.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liballoc-24699c1ddb055eb0.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/librustc_unicode-844a33a197b559a5.rlib /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libbitflags-da52b57722b7b0f5.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liballoc_system-3e467e865c8fa572.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liblibc-60365c932e50e382.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcore-d9873b515905cac5.rlib /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcompiler_builtins-e428224f6caf212a.rlib -l c -s ERROR_ON_UNDEFINED_SYMBOLS=1 -s USE_SDL=2
/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libsdl2-a3439967d6cd896a.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libsdl2_sys-6a6cdeed38de6269.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/liblazy_static-5e13a0eed191ed30.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/librand-35167bebb3b301e7.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/liblibc-09f884e06faabf22.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum-351b1cddc85503c1.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_iter-618824a5957aaa8f.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_integer-9eb11829964f3c6e.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libnum_traits-b3505e86c388a8aa.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libstd-17342542cc541012.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libpanic_unwind-f78756b576499725.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libunwind-11f7709e0c71505b.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/librand-46ed9b788a6928f6.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcollections-e32369d7fef31fbf.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liballoc-24699c1ddb055eb0.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/librustc_unicode-844a33a197b559a5.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/projects/gwar/target/asmjs-unknown-emscripten/debug/deps/libbitflags-da52b57722b7b0f5.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liballoc_system-3e467e865c8fa572.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/liblibc-60365c932e50e382.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcore-d9873b515905cac5.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered

/Users/hoster/emsdk_portable/clang/fastcomp/build_incoming_64/bin/llvm-nm: /Users/hoster/.multirust/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/asmjs-unknown-emscripten/lib/libcompiler_builtins-e428224f6caf212a.rlib(rust.metadata.bin) The end of the file was unexpectedly encountered
```

**Comment by brson** [comment]
Looks like the first issue was fixed. The remaining issue seems to be the same as https://github.com/rust-lang/rust/issues/38558 so closing this one.
